Source/Core/Common/Config/ConfigInfo.h

```diff
@@ -35,80 +35,73 @@ template <typename T>
 class Info
 {
 public:
-  constexpr Info(const Location& location, const T& default_value)
-      : m_location{location}, m_default_value{default_value}, m_cached_value{default_value, 0}
+  constexpr Info(Location location, T default_value)
+      : m_location{std::move(location)}, m_default_value{default_value},
+        m_cached_value{std::move(default_value), 0}
   {
   }
 
-  Info(const Info<T>& other) { *this = other; }
-
-  // Not thread-safe
-  Info(Info<T>&& other) { *this = std::move(other); }
-
-  // Make it easy to convert Info<Enum> into Info<UnderlyingType<Enum>>
-  // so that enum settings can still easily work with code that doesn't care about the enum values.
-  template <Common::TypedEnum<T> Enum>
-  Info(const Info<Enum>& other)
+  Info(const Info<T>& other)
+      : m_location{other.m_location}, m_default_value{other.m_default_value},
+        m_cached_value(other.GetCachedValue())
   {
-    *this = other;
   }
 
-  Info<T>& operator=(const Info<T>& other)
+  Info(Info&& other)
+      : m_location{std::move(other.m_location)}, m_default_value{std::move(other.m_location)},
+        m_cached_value(other.GetCachedValue())
```

Comment:
This move constructor never gets called; if it did then initializing `m_default_value` with `other.m_location` would cause a compile error.

Reply:
Oops. Good catch on `m_default_value`. I suppose I should probably just remove the move constructor.